src/db.c

```diff
@@ -107,7 +107,7 @@ robj *lookupKeyReadWithFlags(redisDb *db, robj *key, int flags) {
 
     if (expireIfNeeded(db,key) == 1) {
         /* Key expired. If we are in the context of a master, expireIfNeeded()
-         * returns 0 only when the key does not exist at all, so it's safe
+         * returns 1 only when the key does not exist at all, so it's safe
```

Comment:
I think the code has evolved past the original implementation, because the comment doesn't really make sense. 
```
If we are in the context of a master, expireIfNeeded() returns 1
when the key is no longer valid, so we can return NULL ASAP.
```

Reply:
updated.